api/taskcluster/webhook.go

```diff
@@ -63,47 +151,20 @@ func tcStatusWebhookHandler(w http.ResponseWriter, r *http.Request) {
 	log.Debugf("GitHub Delivery: %s", r.Header.Get("X-GitHub-Delivery"))
 
 	aeAPI := shared.NewAppEngineAPI(ctx)
-	var status statusEventPayload
-	if err := json.Unmarshal(payload, &status); err != nil {
-		log.Errorf("%v", err)
-		http.Error(w, err.Error(), http.StatusBadRequest)
-		return
-	}
 
-	var processed bool
-	if !shouldProcessStatus(log, &status) {
-		processed = false
-	} else {
-		processed, err = func() (bool, error) {
-			sha := *status.SHA
+	event, err := GetEventInfo(payload, log, &apiImpl{})
 
-			if status.TargetURL == nil {
-				return false, errors.New("No target_url on taskcluster status event")
-			}
-			rootURL, taskGroupID, taskID := parseTaskclusterURL(*status.TargetURL)
-			if taskGroupID == "" {
-				return false, fmt.Errorf("unrecognized target_url: %s", *status.TargetURL)
-			}
-
-			log.Debugf("Taskcluster task group %s", taskGroupID)
-
-			labels := mapset.NewSet()
-			if status.IsOnMaster() {
-				labels.Add(shared.MasterLabel)
-			}
-			sender := status.GetCommit().GetAuthor().GetLogin()
-			if sender != "" {
-				labels.Add(shared.GetUserLabel(sender))
-			}
-
-			log.Debugf("Taskcluster task group %s", taskGroupID)
-			group, err := getTaskGroupInfo(rootURL, taskGroupID)
-			if err != nil {
-				return false, err
-			}
+	processed := false
```

Comment:
Again, I think we should first check `if err != nil` before doing anything with the value. If you check err != nil, its safe to assume that event is not nil and remove `if event != nil` check

Reply:
Done, albeit I'm not happy where we end up. We're back in the realm of a far too large (imo) if/else switch, see https://github.com/web-platform-tests/wpt.fyi/pull/1971/files#diff-5ed03f025b533061a381c9d5a256ae2cR201

I don't understand why the creators of golang made their language so adversarial to proper abstraction -_-

Reply:
(Also we lose a bunch of tests, which makes me sad...)

Reply:
Yea we cannot help it unless we pass in `http` into the function, which makes it harder to keep track of the return statuses.